### PR TITLE
add http ping support

### PR DESCRIPTION
**Pull request by yonghua-sun**

### CODE REVIEW

**Review by EvanLjp — CHANGES_REQUESTED** [review]

File: plugins/input/netping/netping.go [hunk 1]
```diff
@@ -165,60 +190,61 @@ func (m *NetPing) Init(context ilogtail.Context) (int, error) {
...
+					c.Name = fmt.Sprintf("%s -> %s", c.Src, c.Target)
+				default:
+					c.Name = fmt.Sprintf("%s -> %s://%s", c.Src, u.Scheme, u.Host)
+				}
+			}
+
+			localHTTPConfigs = append(localHTTPConfigs, c)
+			m.hasConfig = true
+			if !m.DisableDNS {
+				realHost := strings.Split(u.Host, ":")
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:228]: modified

File: plugins/input/netping/netping.go [hunk 2]
```diff
@@ -244,26 +270,35 @@ func (m *NetPing) Collect(collector ilogtail.Collector) error {
...
+			} else {
+				helper.AddMetric(collector, "dns_resolve_success", nowTs, result.Label, 0)
+				helper.AddMetric(collector, "dns_resolve_failed", nowTs, result.Label, 1)
 			}
-			helper.AddMetric(collector, "dns_resolve_success", nowTs, result.Label, float64(successCount))
 		}
 	}
 
 	counter := 0
 	if len(m.ICMPConfigs) > 0 {
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go]: modified

File: plugins/input/netping/netping.go [hunk 3]
```diff
@@ -244,26 +270,35 @@ func (m *NetPing) Collect(collector ilogtail.Collector) error {
 
 		for i := 0; i < resolveCounter; i++ {
 			result := <-m.resolveChannel
-			successCount := 0
 			if result.Success {
-				successCount++
 				helper.AddMetric(collector, "dns_resolve_rt_ms", nowTs, result.Label, result.RTMs)
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:275]: add failed metric is for query convenient

File: plugins/input/netping/netping.go [hunk 4]
```diff
@@ -244,26 +270,35 @@ func (m *NetPing) Collect(collector ilogtail.Collector) error {
...
+				helper.AddMetric(collector, "dns_resolve_success", nowTs, result.Label, 1)
+				helper.AddMetric(collector, "dns_resolve_failed", nowTs, result.Label, 0)
+			} else {
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:278]: add failed metric is for query convenient

File: plugins/input/netping/netping.go [hunk 5]
```diff
@@ -274,30 +309,97 @@ func (m *NetPing) Collect(collector ilogtail.Collector) error {
 
 	for i := 0; i < counter; i++ {
 		result := <-m.resultChannel
+
+		if !result.Valid {
+			continue
+		}
+
 		helper.AddMetric(collector, fmt.Sprintf("%s_total", result.Type), nowTs, result.Label, float64(result.Total))
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:312]: add failed metric is for query convenient

File: plugins/input/netping/netping.go [hunk 6]
```diff
@@ -274,30 +309,97 @@ func (m *NetPing) Collect(collector ilogtail.Collector) error {
...
+	if exists {
+		realTarget = v.(string)
+	}
+	if realTarget == "" {
+		realTarget = target
+	}
+	return realTarget
+}
+
+func (m *NetPing) doICMPing(config *ICMPConfig) {
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:381]: modified

File: plugins/input/netping/netping.go [hunk 7]
```diff
@@ -345,7 +443,33 @@ func (m *NetPing) doICMPing(config ICMPConfig) {
...
+
+	return time.Since(now), nil
+}
+
+func (m *NetPing) doTCPing(config *TCPConfig) {
+	// prepare labels
+	var label helper.KeyValues
+	label.Append("name", config.Name)
+	label.Append("src", config.Src)
+	label.Append("dst", config.Target)
```
[the remark this answers is not in the document]
Comment (yonghua-sun) [reply plugins/input/netping/netping.go:466]: modified

**Pull request merged**